open-craft/opencraft
# [SE-3362] Send periodic build failure notifications

Make sure to send periodic build failure notifications if periodic builds are enabled on the instance.

**JIRA tickets**: https://openedx.atlassian.net/browse/BTR-8

**Screenshots**: 
N/A

**Sandbox URL**:
N/A

**Testing instructions**:

1. Create an instance
2. Enable periodic builds on the instance
3. Add your email address to the Periodic build notification emails text field
4. Set `00:05:00` as build interval (5 minutes)
5. Misconfigure it to no be able to provision (for example change the configuration repo branch to `notexistsatall`)
6. Wait for the email (check your spam folder as well) - It should arrive in 5-10 minutes based on the delay in the queue

**Author notes and concerns**:

1. The testing of this functionality is time consuming. In case you would like to check logs instead of emails, you should look out for the "`Sending notification e-mail to %s after instance %s didn't provision`" log line.

**Reviewers**
